Samples/iOS-Swift/iOS-Swift/Tools/UIAssert.swift

```diff
@@ -0,0 +1,95 @@
+        if isFailed {
+            return
+        }
+        
+        view.message = success ? "ASSERT: SUCCESS" : "ASSERT: FAIL"
+        view.errorMessage = success ? "" : errorMessage
+        isFailed = !success
+        
+        guard var targetViewController = UIApplication.shared.delegate?.window??.rootViewController else { return }
+        
+        while let presented = targetViewController.presentedViewController {
+            targetViewController = presented
+        }
+        
+        guard let targetView = targetViewController.view else { return }
+        
+        if view.superview != targetView {
+            view.removeFromSuperview()
+            
+            targetView.addSubview(view)
+            
+            let constraints = [
+                view.leftAnchor.constraint(equalTo: targetView.leftAnchor, constant: 0),
+                view.rightAnchor.constraint(equalTo: targetView.rightAnchor, constant: 0),
+                view.bottomAnchor.constraint(equalTo: targetView.bottomAnchor, constant: 0)
+            ]
+            NSLayoutConstraint.activate(constraints)
+        }
+    }
+    
+    func reset() {
+        isFailed = false
+    }
+    
+    static func isTrue(_ value: Bool, _ errorMessage: String? = nil) {
+        shared.assert(success: value, errorMessage: errorMessage)
+    }
+    
+    static func isFalse(_ value: Bool, _ errorMessage: String? = nil) {
+        shared.assert(success: !value, errorMessage: errorMessage)
+    }
+    
+    static func isEqual<T>(_ first: T, _ second: T, _ errorMessage: String? = nil)  where T: Equatable {
+        shared.assert(success: first == second, errorMessage: errorMessage)
+    }
+        
+    static func notNil(_ value: Any?, _ errorMessage: String? = nil) {
+        shared.assert(success: value != nil, errorMessage: errorMessage)
+    }
+    
+    static func isNil(_ value: Any?, _ errorMessage: String? = nil) {
+        shared.assert(success: value == nil, errorMessage: errorMessage)
+    }
+    
+    static func fail(_ errorMessage: String? = nil) {
+        shared.assert(success: false, errorMessage: errorMessage)
+    }
+    
+    static func hasViewControllerLifeCycle(_ span: Span, _ viewController: String) {
+        guard let children = span.children() else {
+            shared.assert(success: false, errorMessage: "\(viewController) span has no children")
+            return
+        }
+        
+        let loadViewSpan = children.first { $0.context.spanDescription == "loadView" }
+        let viewDidLoadSpan = children.first { $0.context.spanDescription == "viewDidLoad" }
+        let viewWillAppearSpan = children.first { $0.context.spanDescription == "viewWillAppear" }
+        let viewDidAppearSpan = children.first { $0.context.spanDescription == "viewDidAppear" }
+        let viewAppearingSpan = children.first { $0.context.spanDescription == "viewAppearing" }
+        
+        notNil(loadViewSpan, "\(viewController) has no loadView span")
```

Comment:
`m`: Can we also add validation for subviews, if the controller should have one?
<img width="231" alt="Screen Shot 2021-12-06 at 12 02 57" src="https://user-images.githubusercontent.com/2443292/144835212-64441c36-d7a3-457f-a687-f81c755dc2ea.png">

Reply:
We have a problem here.
Without animation Sentry SDK don't track layout methods
See #1506 

Of course, we can use animation to present view controllers during test.

Reply:
I think some of our ViewControllers have spans for subviews. We could make the validation configurable.